client/src/BidList.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { get, post } from "./TenderClient";
+
+const BuyerTenderList = () => {
+	const { pageNumber } = useParams();
+	const { tenderId } = useParams();
+	const currentPage = pageNumber ? parseInt(pageNumber, 10) : 1;
+	const [bid, setBid] = useState([]);
+	const [updateStatus, setUpdatedStatus] = useState({
+		id: 1,
+		status: "Active",
+	});
+	const [errorMsg, setErrorMsg] = useState(null);
+	const [loading, setLoading] = useState(true);
+	const navigate = useNavigate();
+	const [pagination, setPagination] = useState({
+		itemsPerPage: 10,
+		currentPage: currentPage,
+		totalPages: 1,
+	});
+
+	const fetchBids = async (tenderId, page) => {
+		setLoading(true);
+		try {
+			const data = await get(`/api/bid?tender_id=${tenderId}&page=${page}`);
+			setBid(data.results);
+			setPagination(data.pagination);
+			setErrorMsg(null);
+		} catch (error) {
+			setErrorMsg("Error fetching bids!");
+		} finally {
+			setLoading(false);
+		}
+	};
+
+	useEffect(() => {
+		fetchBids(tenderId, currentPage);
+	}, [currentPage, tenderId]);
+
+	const handleBidStatusChange = async (bidId, status) => {
+		const response = await post(`/api/bid/${bidId}/status`, { status });
+		if (response.code === "SUCCESS") {
+			setUpdatedStatus({ id: bidId, status: status });
+		} else {
+			setErrorMsg("Error fetching bids!");
+		}
+	};
+
+	if (bid.length === 0) {
+		return <div className="msg">No Bid Submited yet!!</div>;
+	}
+
+	const loadNextPage = () => {
+		if (pagination.currentPage < pagination.totalPages && !loading) {
+			navigate(`/bidding/page/${pagination.currentPage + 1}`);
+		}
+	};
+
+	const loadPreviousPage = () => {
+		if (pagination.currentPage > 1 && !loading) {
+			navigate(`/bidding/page/${pagination.currentPage - 1}`);
+		}
+	};
+
+	if (errorMsg !== null) {
+		return <div>{errorMsg}</div>;
+	}
+
+	if (loading) {
+		return <div>Loading!!</div>;
+	}
+
+	return (
+		<main>
+			<h2 className="heading">Bids</h2>
+			<div className="container">
+				{errorMsg && <p className="error-message">{errorMsg}</p>}
+				{bid.map((bid) => (
+					<div className="card" key={bid.bid_id}>
+						<p className="posted-on">
+							Submited on
+							<span className="posted-on-date">
+								{new Date(bid.bidding_date).toLocaleDateString()}
+							</span>
+							<span className="bid-status">
+								{updateStatus.id === bid.bid_id
+									? updateStatus.status
+									: bid.status}
```

Comment:
This is kind of complicated for what you are trying to do. I would just show a message telling the user that a specific bid was accepted or rejected and then re-fetch the current page